src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using static CharacterController;
+
+public class DeckController : MonoBehaviour
+{
+    [SerializeField] private PartyDecks decks;
+
+    private Deck current;
+
+    public void OnHeroChange()
+    {
+        this.current = decks.Decks[
+            Array.IndexOf(
+                Enum.GetValues(CharacterController.currentCharacter.GetType()),
```

Comment:
We do not want to depend on a non-state type.

DeckBuilder State should live in a DeckBuilderState scriptable object. 

If there is to be a current character, it should be there.